src/tree/updater_quantile_hist.cc

```diff
@@ -271,6 +279,25 @@ void QuantileHistMaker::Builder::EvaluateSplits(
   }
 }
 
+
+void QuantileHistMaker::Builder::EvaluateSplits(
+    const GHistIndexMatrix &gmat,
+    const ColumnMatrix &column_matrix,
+    DMatrix *p_fmat,
+    RegTree *p_tree,
+    int *num_leaves,
+    int depth,
+    unsigned *timestamp,
+    std::vector<ExpandEntry> *temp_qexpand_depth) {
+  EvaluateSplit(qexpand_depth_wise_, gmat, hist_, *p_fmat, *p_tree);
+
+  std::vector<ExpandEntry> nodes_for_apply_split;
+  AddSplitsToTree(gmat, p_fmat, p_tree, num_leaves, depth, timestamp,
```

Comment:
Please remove the function `EvaluateSplits` as it's doing multiple things, including evaluation and application.

Reply:
I'm not so strong in naming, so renamed:
EvaluateSplits -> EvaluateAndApplySplits
EvaluateSplit -> EvaluateSplits